src/drr/drrtool.cpp

```diff
@@ -0,0 +1,192 @@
+/* +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+    Copyright (c) 2017 of Haochuan Chen
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published
+    by the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++ */
+#ifdef __PLUMED_HAS_BOOST_SERIALIZATION
+#include "cltools/CLTool.h"
+#include "cltools/CLToolRegister.h"
+#include "config/Config.h"
+#include "core/ActionRegister.h"
+#include "DRR.h"
+#include "tools/Tools.h"
+#include "tools/Units.h"
+#include <boost/archive/binary_iarchive.hpp>
+#include <boost/archive/binary_oarchive.hpp>
+#include <boost/serialization/vector.hpp>
+#include <cstdlib>
+#include <fstream>
+#include <iostream>
+#include <string>
+
+using namespace PLMD;
+using namespace cltools;
+
+namespace PLMD {
+namespace drr {
+
+//+PLUMEDOC TOOLS drrtool
+/*
+ - Extract .grad and .count files from the binary output .drrstate
+ - Merge windows
+
+\par Examples
+
+The following command will extract .grad and .count files.
+\verbatim
+plumed drrtool --extract eabf.drrstate
+\endverbatim
+
+The following command will merge windows of two .drrstate file, and output the
+.grad and .count files.
+\verbatim
+plumed drrtool --merge win1.drrstate,win2.drrstate
+\endverbatim
+
+After getting the .grad and .count file, you can do numerical integration by
+using abf_integrate tool from
+https://github.com/Colvars/colvars/tree/master/colvartools
+\verbatim
+abf_integrate eabf.czar.grad
+\endverbatim
+
+*/
+//+ENDPLUMEDOC
+
+class drrtool : public CLTool {
+public:
+  static void registerKeywords(Keywords &keys);
+  explicit drrtool(const CLToolOptions &co);
+  int main(FILE *in, FILE *out, Communicator &pc);
+  void extractdrr(const std::vector<std::string> &filename);
+  void mergewindows(const std::vector<std::string> &filename);
+  std::string description() const { return "Extract or merge the drrstate files."; }
+
+private:
+  bool verbosity;
+  const std::string suffix{".drrstate"};
+};
+
+PLUMED_REGISTER_CLTOOL(drrtool, "drrtool")
```

Comment:
We are in the process of deciding rules for namespaces. Perhaps you should change this to
````
PLUMED_REGISTER_CLTOOL(drrtool, "drr_tool")
````
(notice  the underscore)